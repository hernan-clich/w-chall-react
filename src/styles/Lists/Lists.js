import styled from 'styled-components';

export const StyledLists = styled.div`
    min-width: 100vw;
    height: auto;

    h2 {
        text-align: center;
        margin-top: 2em;
    }

    #list-wrapper {
        min-width: 100vw;
        height: auto;
    }

    .list-counter-wrapper {
        min-width: 100vw;
        max-height: 50vh;
        min-height: 30vh;
        background-color: rgba(40,40,40,1);
        display: flex;
        justify-content: center;
        align-items: center;

        p {
            font-family: 'Montserrat-SemiBoldItalic';
            color: rgba(225,225,225,1);
            padding: 0 1em;
            text-align: center;
        }
    }

    @media (min-width: 1361px){
        .list-counter-wrapper {
            p {
                font-size: 1.75vw;
            }
        }
    }
`;