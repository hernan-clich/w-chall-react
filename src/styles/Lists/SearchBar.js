import styled from 'styled-components';

import { primary, secondary, plainWhite } from '../Variables';

export const StyledSearchBar = styled.div`
    min-width: 100vw;
    min-height: 70vh;
    display: flex;
    flex-flow: column;
    justify-content: center;
    align-items: center;
    position: relative;

    .list-img {
        min-width: 150px;
        width: 20vw;
        height: 7.5vh;
        margin-bottom: 2.5em;
        background: url('./assets/logo_full_color.svg') no-repeat;
        background-size: 100%;
    }

    .list-search-input {
        min-width: 200px;
        width: 50%;
        position: relative;

        &::after {
            position: absolute;
            content:"";
            width: 20px;
            height: 20px;
            background: url(./assets/lists-search.svg) no-repeat;
            background-size: cover;
            right: 5%;
            top: 50%;
            z-index: 1;
            transform: translateY(-50%);
        }

        input[type="text"] {
            padding: 0.75em 0 0.75em 1.5em;
            width: 100%;
            border-radius: 5px;
            font-size: 1rem;
            border: 1px solid rgba(190,190,190,1);
            transition: box-shadow 250ms ease-in-out;
            font-family: 'Montserrat-SemiBold';

            @media (hover: hover) {
                &:hover {
                    box-shadow: 0px 0px 3px 0px rgba(189,189,189,1);
                }
            }
            
            &:focus {
                outline: none;
                box-shadow: 0px 0px 7px 0px rgba(189,189,189,1);
            }
        }
    }

    .list-radios {
        display: flex;
        flex-flow: row nowrap;
        align-items: center;
        margin-top: 1.5em;

        .single-list-radio {
            display: flex;
            flex-flow: row nowrap;
            align-items: center;
            padding: 0.25em 0;
            margin: 0 0.5em;
            cursor: pointer;

            .list-radio-wrapper {
                width: 0.9em;
                height: 0.9em;
                margin: 0 0.5em;
                background-color: rgba(120, 120,120,1);
                clip-path: circle(50% at 50% 50%);
                position: relative;
                cursor: pointer;

                &::before {
                    position: absolute;
                    content:"";
                    top: 7.5%;
                    right: 7.5%;
                    bottom: 7.5%;
                    left: 7.5%;
                    background-color: ${plainWhite()};
                    clip-path: circle(50% at 50% 50%);
                }

                .list-radio-tick {
                    position: absolute;
                    content:""; 
                    clip-path: circle(0% at 50% 50%);
                    transition: clip-path 500ms ease;
                }
            }

            .radio-ticked.list-radio-wrapper  {
                background-color: ${primary()};

                .list-radio-tick {
                    
                    width: 100%;
                    height: 100%;
                    background-color: ${primary()};
                    clip-path: circle(30% at 50% 50%);
                }
            }

            label {
                font-family: 'Montserrat-Medium';
                font-size: 0.8rem;
                margin-right: 1em;
                cursor: pointer;
            }
        }
    }
    
    .list-sorting {
        display: flex;
        flex-flow: row nowrap;
        justify-content: center;
        align-items: center;
        margin-top: 2em;
        transition: all 250ms ease;

        p.sort-asc, p.sort-des {
            font-family: 'Montserrat-SemiBold';
            font-size: 0.75rem;
            cursor: pointer;
            position: relative;
            padding: 0.5em;

            &::after {
                position: absolute;
                content: "";
                width: 14px;
                height: 14px;
                top: 50%;
                right: -8.5%;
                pointer-events: none;
                transform: translateY(-50%);
                transition: all 150ms ease;
            }
            
            span {
                font-family: 'Montserrat-SemiBoldItalic';
                transition: text-shadow 300ms ease;
            }
        }

        p.sort-asc {
            span {
                color: ${primary()};
            }

            &::after {
                background-color: ${primary()};
                clip-path: polygon(50% 0%, 0% 100%, 100% 100%);
            }

            @media (hover: hover) {
                &:hover {
                    > span {
                        text-shadow: -5px 5px 0px rgba(0, 230, 230, 0.5), -10px 10px 0px rgba(0, 204, 204, 0.3), -15px 15px 0px rgba(0, 189, 189, 0.1), 2px 2px 2px rgba(42,167,223,0);
                    }
                }
            }
        }            

        p.sort-des {
            span {
                color: ${secondary()};
            }

            &::after {
                background-color: ${secondary()};
                clip-path: polygon(50% 100%, 0 0, 100% 0);
            }

            @media (hover: hover) {
                &:hover {
                    > span {
                        text-shadow: -5px 5px 0px rgba(163, 204, 57, 0.5), -10px 10px 0px rgba(163, 204, 57, 0.3), -15px 15px 0px rgba(163, 204, 57, 0.1), 2px 2px 2px rgba(163,204,57,0);
                    }
                }
            }
        }
    }

    @media (min-width: 1361px){
        .list-img {
            margin-bottom: 5.5vh;
        }
        
        .list-search-input {

            &::after {
                width: 2.85%;
                height: 40%;
            }

            input[type="text"] {
                border-radius: 0.25em;
                font-size: 1.25vw;
                border: 0.05em solid rgba(190,190,190,1);

                @media (hover: hover) {
                    &:hover {
                        box-shadow: 0px 0px 0.25em 0px rgba(189,189,189,1);
                    }
                }
                
                &:focus {
                    outline: none;
                    box-shadow: 0px 0px 0.5em 0px rgba(189,189,189,1);
                }
            }
        }

        .list-radios {
            min-width: 5vw;
            height: 5vh;
            margin-top: 3vh;

            .list-radio-wrapper {
                min-width: 8%;
                padding-bottom:8%;
            }

            label {
                font-family: 'Montserrat-Medium';
                font-size: 1vw;
                margin: 0 1em 0 0.25em;
            }
        }

        .list-sorting {

            p.sort-asc, p.sort-des {
                font-size: 0.9vw;
                margin-top: 0;

                &::after {
                    width: 5%;
                    height: 80%;
                }
            }

            p.sort-asc {

                @media (hover: hover) {
                    &:hover {
                        > span {
                            text-shadow: -0.25em 0.25em 0px rgba(0, 230, 230, 0.5), -0.5em 0.5em 0px rgba(0, 204, 204, 0.3), -0.75em 0.75em 0px rgba(0, 189, 189, 0.1), 2px 2px 2px rgba(42,167,223,0);
                        }
                    }
                }
            }            

            p.sort-des {

                @media (hover: hover) {
                    &:hover {
                        > span {
                            text-shadow: -0.25em 0.25em 0px rgba(163, 204, 57, 0.5), -0.5em 0.5em 0px rgba(163, 204, 57, 0.3), -0.75em 0.75em 0px rgba(163, 204, 57, 0.1), 2px 2px 2px rgba(163,204,57,0);
                        }
                    }
                }
            }
        }
    }
`;