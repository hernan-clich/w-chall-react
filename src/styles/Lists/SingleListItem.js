import styled from 'styled-components';

import { primary, secondary } from '../../styles/Variables';

export const StyledListItem = styled.div`
    min-width: 95vw;
    min-height: 20vh;
    display: flex;
    justify-content: center;
    margin: 0 auto;
    word-wrap: break-word;
    position: relative;

    &::after {
        position: absolute;
        content: "";
        width: 90%;
        top: 0%;
        left: 4%;
        height: 1px;
        background-color: rgba(220,220,220,1);
    }

    .list-top {
        max-height: 20vh;
        width: 50%;
        display: flex;
        justify-content: center;
        align-items: center;
        
        .list-logo {
            height: 3rem;
            width: 3rem;
            border-radius: 10px;
            margin-left: 1em;
            border: 2px solid black;
            flex-shrink: 0;
            background-color: white;
            background-repeat: no-repeat;
            background-position: center center;
            background-size: 80%;
        }

        .list-tech {
            min-height: 100%;
            min-width: 24vw;
            width: auto;
            display: flex;
            flex-flow: column nowrap;
            justify-content: center;
            margin-left: 5%;
            justify-self: flex-start;

            .tech-header {
                font-size: 0.7rem;
                font-family: Montserrat-Bold;
            }

            .tech-subheader {
                display: flex;
                flex-flow: column nowrap;
                justify-content: flex-start;
                font-size: 0.65rem;
                font-family: Montserrat-Bold;
                color: rgba(90,90,90,1);
                padding: 0.2em 0;

                p:nth-child(2) {
                    font-family: Montserrat-SemiBoldItalic;
                    font-size: 0.6rem;
                    padding: 0.1em 0;
                }

                p:nth-child(3) {
                    font-family: Montserrat-SemiBoldItalic;
                    font-size: 0.55rem;
                }
            }
        }
    }

    .list-bottom {
        max-height: 20vh;
        width: 50%;
        display: flex;
        margin-right: 2.5%;

        .list-author {
            min-height: 100%;
            min-width: 25vw;
            display: flex;
            flex-flow: column nowrap;
            align-items: flex-start;
            justify-content: center;
            font-size: 0.6rem;
            line-height: 1.4;
            font-family: Montserrat-SemiBold;
            padding-left: 1em;

            p:nth-child(1) {
                font-family: 'Montserrat-SemiBoldItalic';
                font-size: 0.55rem;
            }
        }

        .list-type {
            min-height: 100%;
            min-width: 25vw;
            display: flex;
            flex-flow: column nowrap;
            align-items: flex-start;
            justify-content: center;
            font-size: 0.7rem;
            font-family: Montserrat-Bold;

            .list-type-wrapper {
                margin-left: 0.5em;

                p[data-type='Mobile'] {
                    color: crimson;
                }

                p[data-type='Back-End'] {
                    color: ${secondary()};
                }

                p[data-type='Front-End'] {
                    color: ${primary()};
                }
            }
        }
    }

    @media (min-width: 576px){
        .list-top {
            
            .list-logo {
                height: 4rem;
                width: 4rem;
            }

            .list-tech {

                .tech-header {
                    font-size: 1rem;
                }

                .tech-subheader {
                    font-size: 0.8rem;
                    padding: 0.2em 0;

                    p:nth-child(2) {
                        font-size: 0.7rem;
                        padding: 0.3em 0;
                    }

                    p:nth-child(3) {
                        font-size: 0.6rem;
                    }
                }
            }
        }

        .list-bottom {

            .list-author {
                font-size: 0.8rem;
                padding-left: 1em;

                p:nth-child(1) {
                    font-family: 'Montserrat-SemiBoldItalic';
                    font-size: 0.7rem;
                }
            }

            .list-type {
                font-size: 0.9rem;
            }
        }
    }

    @media (min-width: 1361px){
        &::after {
            height: 0.1em;
        }

        .list-top {
            
            .list-logo {
                height: 50%;
                width: 10%;
                border-radius: 0.75em;
                border: 0.2em solid black;
            }

            .list-tech {

                .tech-header {
                    font-size: 1.25vw;
                }

                .tech-subheader {
                    font-size: 0.9vw;

                    p:nth-child(2) {
                        font-size: 0.8vw;
                    }

                    p:nth-child(3) {
                        font-size: 0.75vw;
                    }
                }
            }
        }

        .list-bottom {

            .list-author {
                font-size: 0.95vw;

                p:nth-child(1) {
                    font-size: 0.8vw;
                }
            }

            .list-type {
                font-size: 1.1vw;
            }
        }
    }
`;