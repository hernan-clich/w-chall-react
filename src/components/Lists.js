import React, { useState } from 'react';
import { Redirect } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { useQuery } from 'react-query';

import Header from './Header';
import { StyledLists } from '../styles/Lists/Lists';
import SingleListItem from './Lists/SingleListItem';

const fetchTecnologias = async () => {
  const res = await fetch('http://private-8e8921-woloxfrontendinverview.apiary-mock.com/techs');
  return res.json();
};

const Lists = () => {
  const [searchList, setSearchList] = useState('');
  const [sortAscending, setSortAscending] = useState(true);
  const [searchByName, setSearchByName] = useState(true);

  const isUserLoggedIn = useSelector(state => state.isSignedIn);

  const { data, status } = useQuery('tecnologias', fetchTecnologias);

  const sortList = arr => {
    if(sortAscending){
      return arr.sort((a, b) => (a.tech.toLowerCase() > b.tech.toLowerCase()) ? 1 : -1);
    } else {
      return arr.sort((a, b) => (a.tech.toLowerCase() < b.tech.toLowerCase()) ? 1 : -1);
    }
  };

  if (!isUserLoggedIn && !localStorage.getItem('user')) {
    return <Redirect to="/login"/>;
  }

  return (
    <>
      <Header/>
      <StyledLists>
        <div id="list-search-wrapper">
          <div className="list-search-input">
            <input 
              type="text" 
              name="search-list" 
              id="search-list"
              onChange={e => setSearchList(e.target.value)}
              value={searchList}
            />
          </div>
          <div className="list-radios" >
            <input 
              type="radio" 
              value="Nombre" 
              id="nombre"
              onChange={() => setSearchByName(!searchByName)} 
              checked={searchByName} 

            />
            <label htmlFor="tipo">Nombre</label>
            <input 
              type="radio" 
              value="Tipo" 
              id="tipo"
              onChange={() => setSearchByName(!searchByName)} 
              checked={!searchByName} 
            />
            <label htmlFor="tipo">Tipo</label>
          </div>
          <div className="list-sorting">
            <p 
              onClick={() => setSortAscending(!sortAscending)}
              className={sortAscending ? 'ascendente' : 'descendente'}
            >
              Ordenando de forma <span>{sortAscending ? 'ascendente' : 'descendente'}</span>
            </p>
          </div>
        </div>
        {status === 'loading' && (
          <h2>Cargando listado...</h2>
        )}
        {status === 'success' && (
          <>
            <div id="list-wrapper">
              {
                sortList(data.filter(tech => tech.tech.toLowerCase().includes(searchList.toLowerCase())))
                  .map(tech => (
                    <SingleListItem
                      key={tech.tech}
                      logo={tech.logo}
                      tech={tech.tech}
                      language={tech.language}
                      year={tech.year}
                      author={tech.author}
                      license={tech.license}
                      type={tech.type}
                    />
                  ))
              }
            </div>
            <div className="list-counter-wrapper">
              <p>Actualmente viendo: {data.filter(tech => tech.tech.toLowerCase().includes(searchList.toLowerCase())).length} elementos</p>
            </div>
          </>
        )}
      </StyledLists>
    </>
  );
};

export default Lists;
