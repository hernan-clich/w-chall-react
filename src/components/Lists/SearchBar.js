import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import { StyledSearchBar } from '../../styles/Lists/SearchBar';
import { FooterButton } from '../Buttons';

const SearchBar = props => {
  
  return (
    <StyledSearchBar>
      <FooterButton as={Link} to="/" className="list-button-back">Volver a Home</FooterButton>
      <div className="list-img"/>
      <div className="list-search-input">
        <input 
          type="text" 
          name="search-list" 
          id="search-list"
          onChange={e => props.setSearchList(e.target.value)}
          value={props.searchList}
        />
      </div>
      <div className="list-radios" >
        <input 
          type="radio" 
          value="Nombre" 
          id="nombre"
          onChange={() => props.setSearchByName(!props.searchByName)} 
          checked={props.searchByName} 

        />
        <label htmlFor="tipo">Nombre</label>
        <input 
          type="radio" 
          value="Tipo" 
          id="tipo"
          onChange={() => props.setSearchByName(!props.searchByName)} 
          checked={!props.searchByName} 
        />
        <label htmlFor="tipo">Tipo</label>
      </div>
      <div className="list-sorting">
        <p 
          onClick={() => props.setSortAscending(!props.sortAscending)}
          className={props.sortAscending ? 'sort-asc' : 'sort-des'}
        >
              Ordenando de forma <span>{props.sortAscending ? 'ascendente' : 'descendente'}</span>
        </p>
      </div>
    </StyledSearchBar>
  );
};

SearchBar.propTypes = {
  searchList: PropTypes.string.isRequired,
  setSearchList: PropTypes.func.isRequired,
  sortAscending: PropTypes.bool.isRequired,
  setSortAscending: PropTypes.func.isRequired,
  searchByName: PropTypes.bool.isRequired,
  setSearchByName: PropTypes.func.isRequired
};

export default SearchBar;