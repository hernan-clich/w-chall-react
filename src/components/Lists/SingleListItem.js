import React from 'react';
import PropTypes from 'prop-types';

import { StyledListItem } from '../../styles/Lists/SingleListItem';

const SingleListItem = props => {

  const techieLogo = {backgroundImage: `url(${props.logo})`};
  
  return (
    <StyledListItem>
      <div className="list-top">
        <div style={techieLogo} className="list-logo"/>
        <div className="list-tech">
          <div className="tech-header">
            <p>{props.tech}</p>
          </div>
          <div className="tech-subheader">
            <p>{props.language}</p>
            <p>Licencia: {props.license}</p>
            <p>Lanzamiento: {props.year}</p>
          </div>
        </div>
      </div>
      <div className="list-bottom">
        <div className="list-author">
          <p>Autor:</p>
          <p>{props.author}</p>
        </div>
        <div className="list-type">
          <div className="list-type-wrapper">
            <p data-type={props.type}>{props.type}</p>
          </div>
        </div>
      </div>
    </StyledListItem>
  );
};

SingleListItem.propTypes = {
  logo: PropTypes.string, 
  tech: PropTypes.string, 
  language: PropTypes.string, 
  year: PropTypes.string, 
  author: PropTypes.string, 
  license: PropTypes.string, 
  type: PropTypes.string
};

export default SingleListItem;
