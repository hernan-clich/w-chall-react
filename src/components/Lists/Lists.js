import React, { useState } from 'react';
import { Redirect } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { useQuery } from 'react-query';

import { StyledLists } from '../../styles/Lists/Lists';
import SearchBar from './SearchBar';
import SingleListItem from './SingleListItem';

// Fetching data from API
const fetchTecnologias = async () => {
  const res = await fetch('http://private-8e8921-woloxfrontendinverview.apiary-mock.com/techs');
  return res.json();
};

const Lists = () => {

  // Internal state
  const [searchList, setSearchList] = useState('');
  const [sortAscending, setSortAscending] = useState(true);
  const [searchByName, setSearchByName] = useState(true);

  // Bringing state from redux
  const isUserLoggedIn = useSelector(state => state.isSignedIn);

  // React Query hook
  const { data, status } = useQuery('tecnologias', fetchTecnologias);
  
  // Helper functions

  const filterList = arr => {
    if(searchByName){
      return arr.filter(tech => tech.tech.toLowerCase().includes(searchList.toLowerCase()));
    } else {
      return arr.filter(tech => tech.type.toLowerCase().includes(searchList.toLowerCase()));
    }
  };
    
  const sortList = arr => {
    if(sortAscending){
      return arr.sort((a, b) => (a.tech.toLowerCase() > b.tech.toLowerCase()) ? 1 : -1);
    } else {
      return arr.sort((a, b) => (a.tech.toLowerCase() < b.tech.toLowerCase()) ? 1 : -1);
    }
  };

  // If user isn't logged in then redirect to login page

  if (!isUserLoggedIn && !localStorage.getItem('user')) {
    return <Redirect to="/login"/>;
  }

  return (
    <StyledLists>
      <SearchBar
        searchList={searchList}
        setSearchList={setSearchList}
        sortAscending={sortAscending}
        setSortAscending={setSortAscending}
        searchByName={searchByName}
        setSearchByName={setSearchByName}

      />
      {status === 'loading' && (
        <h2>Cargando listado...</h2>
      )}
      {status === 'success' && (
        <>
          <div id="list-wrapper">
            { sortList(filterList(data))
              .map(tech => (
                <SingleListItem
                  key={tech.tech}
                  logo={tech.logo}
                  tech={tech.tech}
                  language={tech.language}
                  year={tech.year}
                  author={tech.author}
                  license={tech.license}
                  type={tech.type}
                />
              ))
            }
          </div>
          <div className="list-counter-wrapper">
            {filterList(data).length > 0 ?
              <p>Actualmente visualizando: {filterList(data).length} elementos</p> :
              <p>Ninguna tecnolog√≠a encontrada con los criterios solicitados</p>
            }
          </div>
        </>
      )}
    </StyledLists>
  );
};

export default Lists;
