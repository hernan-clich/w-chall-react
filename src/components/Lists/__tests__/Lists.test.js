/* eslint-disable no-undef */
import React from 'react';
import Lists from '../Lists';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';


describe('Lists component test', ()=>{
  test('Snapshot of Lists', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Lists/>
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });    

  test('Finding title with TestId', () =>{
    const { getByTestId } = render(
      <BrowserRouter>
        <Lists/>
      </BrowserRouter>
    );
    expect(getByTestId('tipo')).toHaveTextContent(/^Nombre$/);
  });

  test('Check if p tag text changes after click', () =>{
    const { getByTestId } = render(
      <BrowserRouter>
        <Lists/>
      </BrowserRouter>
    );
    const pTag = getByTestId('list-sorting');
    userEvent.click(pTag);
    expect(pTag).toHaveTextContent('Ordenando de forma descendente');
  });
});