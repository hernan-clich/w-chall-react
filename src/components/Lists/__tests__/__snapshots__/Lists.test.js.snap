// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists component test Snapshot of Lists 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx isvsjk"
  >
    <div
      class="sc-gtssRu jpGsWG"
    >
      <a
        class="sc-dlnjPT sc-gKAblj sc-iCoHVE btVOhm iYvDKD dfaScl"
        href="/"
      >
        Volver a Home
      </a>
      <div
        class="list-img"
      />
      <div
        class="list-search-input"
      >
        <input
          id="search-list"
          name="search-list"
          type="text"
          value=""
        />
      </div>
      <div
        class="list-radios"
      >
        <div
          class="single-list-radio"
        >
          <div
            class="list-radio-wrapper radio-ticked"
            id="nombre"
          >
            <div
              class="list-radio-tick"
            />
          </div>
          <label
            data-testid="tipo"
            for="tipo"
          >
            Nombre
          </label>
        </div>
        <div
          class="single-list-radio"
        >
          <div
            class="list-radio-wrapper "
            id="tipo"
          >
            <div
              class="list-radio-tick"
            />
          </div>
          <label
            for="tipo"
          >
            Tipo
          </label>
        </div>
      </div>
      <div
        class="list-sorting"
      >
        <p
          class="sort-asc"
          data-testid="list-sorting"
        >
          Ordenando de forma 
          <span>
            ascendente
          </span>
        </p>
      </div>
    </div>
    <div
      class="sc-kEqYlL cHyWym"
    />
  </div>
</DocumentFragment>
`;
